libraries/AP_ToshibaCAN/AP_ToshibaCAN.cpp

```diff
@@ -539,4 +552,53 @@ void AP_ToshibaCAN::send_esc_telemetry_mavlink(uint8_t mav_chan)
     }
 }
 
+//
+// accessors for latest data from esc
+//
+// return latest rpm
+uint16_t AP_ToshibaCAN::get_rpm(uint8_t esc_id) const
+{
+    if (esc_id > TOSHIBACAN_MAX_NUM_ESCS) {
+        return 0;
+    }
+    return _telemetry[esc_id].rpm;
+}
+
+// return voltage in volts
+float AP_ToshibaCAN::get_voltage(uint8_t esc_id) const
+{
+    if (esc_id > TOSHIBACAN_MAX_NUM_ESCS) {
```

Comment:
Bad range check, needs to be `>=` as you are indexing directly into the array. (If scripts eventually gets to this then maybe we want to index from 1 instead of 0?

Reply:
thanks, fixed!

Reply:
hmm.. sounds like we have a different convention for indices in lua vs C++.  I don't know what we should do about this but at the C++ level we should keep it as starting at 0 I think.  Is it a specific problem to have it start from 0 when using Lua? .. or just not convention?

Reply:
Lua indexes by default from 1 on arrays, but it's not really a problem. It's more kinda how we talk about motors/rc/servo channesl in general in parameters. IE "Channel 1" is actually index 0 to the arrays, and all our servo/rc code tends to reflect this reality. I'm not fussed either way, except to note that maybe this should look like the servo/rc channel in indexing. (It's possible more of those are actually 0 indexed then I recall off the top of my head).